src/components/NewTimerModal.tsx

```diff
@@ -10,95 +10,101 @@ import {
   Button
 } from '@chakra-ui/react'
 import { AddIcon } from '@chakra-ui/icons'
-import { useState, useEffect } from 'react'
-import { Timer, NewTimerModalProps } from '../types/app-types'
+import { useEffect, useState, } from 'react'
+import { NewTimerModalProps, IncomingTimerProps, TimerConfig } from '../types/app-types'
 import NamingPhase from './ModalSteps/NamingPhase'
 import GamingPhase from './ModalSteps/GamingPhase'
 import TimingPhase from './ModalSteps/TimingPhase'
+import { GameKey } from './Timer/timerStyles'
 
 enum Phase {
   SetName,
   SetGameType,
-  SetCountdown
+  SetCountdown,
+  Complete
+}
+
+const initialParams = {
+  name: '',
+  game: '' as GameKey,
+  initialTime: { minutes: 0, seconds: 0 },
 }
 
 const NewTimerModal: React.FC<NewTimerModalProps> = ({ onModalComplete }) => {
   const { isOpen, onOpen, onClose } = useDisclosure()
   const [phase, setPhase] = useState<Phase>(Phase.SetName)
-  const [newTimerParams, setNewTimerParams] = useState<Timer>({
-    title: '',
-    game: '',
-    initialTime: '',
-    timeAtPause: ''
-  })
-
-  const initialTimerParams = {
-    title: '',
-    game: '',
-    initialTime: '',
-    timeAtPause: ''
+  const [newTimerParams, setNewTimerParams] = useState<TimerConfig>(initialParams)
+
+  const resetModal = () => {
+    onClose()
+    setPhase(Phase.SetName)
+    setNewTimerParams(initialParams)
+  }
+
+  const onBack = () => {
+    if (phase === Phase.SetName) {
+      return
+    }
+    setPhase((prevPhase) => prevPhase - 1)
   }
 
-  const goBackOnePhase = () => {
-    if (phase > 0) {
-      console.log('goBackOnePhase firing')
-      setPhase((prevPhase) => prevPhase - 1)
-    } else {
-      return null
+  const nextPhase = () => {
+    const transitions: Partial<Record<Phase, Phase>> = {
+      [Phase.SetName]: Phase.SetGameType,
+      [Phase.SetGameType]: Phase.SetCountdown,
+      [Phase.SetCountdown]: Phase.Complete
+    }
+
+    const nextPhase = transitions[phase]
+    if (nextPhase) {
+      return setPhase(nextPhase)
     }
   }
 
+  useEffect(() => {
+    if (phase !== Phase.Complete) {
+      return
+    }
+
+    const complete = newTimerParams.name && newTimerParams.game && newTimerParams.initialTime
+    if (!complete) {
+      return
+    }
+
+    onModalComplete(newTimerParams)
+    resetModal()
+  }, [phase])
+
+  const onClickNext = (values: Partial<IncomingTimerProps>) => {
```

Comment:
this interface means all phase components can pass part of the final timer config, and we can easily get it into this local state